src/dotnet/MsbuildProject.cs

```diff
@@ -166,12 +174,97 @@ public static void EnsureAllReferencesExist(List<string> references)
             }
         }
 
+        public IEnumerable<NuGetFramework> GetTargetFrameworks()
+        {
+            if (_cachedTfms != null)
+            {
+                return _cachedTfms;
+            }
+
+            var project = GetEvaluatedProject();
+
+            var properties = project.AllEvaluatedProperties
+                                    .Where(p => p.Name.Equals("TargetFrameworks", StringComparison.OrdinalIgnoreCase) ||
+                                                p.Name.Equals("TargetFramework", StringComparison.OrdinalIgnoreCase))
+                                    .Select(p => p.EvaluatedValue.ToLower()).ToList();
+
+            var uniqueTfms = new HashSet<string>();
+
+            foreach (var property in properties)
+            {
+                var tfms = property
+                                .Split(';')
+                                .Select((tfm) => tfm.Trim())
+                                .Where((tfm) => !string.IsNullOrEmpty(tfm));
+
+                foreach (var tfm in tfms)
+                {
+                    uniqueTfms.Add(tfm);
+                }
+            }
+
+            _cachedTfms = uniqueTfms.Select((frameworkString) => NuGetFramework.Parse(frameworkString)).ToList();
+            return _cachedTfms;
+        }
+
+        public bool CanWorkOnFramework(NuGetFramework framework)
+        {
+            foreach (var tfm in GetTargetFrameworks())
+            {
+                if (DefaultCompatibilityProvider.Instance.IsCompatible(framework, tfm))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public bool TargetsFramework(NuGetFramework framework)
+        {
+            foreach (var tfm in GetTargetFrameworks())
+            {
+                if (framework.Equals(tfm))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private Project GetEvaluatedProject()
+        {
+            if (_cachedEvaluatedProject != null)
+            {
+                return _cachedEvaluatedProject;
+            }
+
+            var loadedProjects = _collection.GetLoadedProjects(ProjectRoot.FullPath);
```

Comment:
I would break the rest of this method into its own method. Just to make this one nicer to read. Something like, cached? return; Otehrwise, get a new one.